src/streaming/TextSourceBuffer.js

```diff
@@ -91,37 +138,33 @@ MediaPlayer.dependencies.TextSourceBuffer = function () {
                 bytes = new Uint8Array(bytes);
                 ccContent=window.UTF8.decode(bytes);
                 try {
-                    result = self.getParser().parse(ccContent);
-                    label = mediaInfo.id;
-                    lang = mediaInfo.lang;
-                    self.getTextTrackExtensions().addTextTrack(self.videoModel.getElement(), result, label, lang, true);
-                    self.eventBus.dispatchEvent({type:MediaPlayer.events.TEXT_TRACK_ADDED});
+                    result = self.getParser(mimeType).parse(ccContent);
+                    createTextTrackFromMediaInfo(result, mediaInfo);
                 } catch(e) {
                     self.errHandler.closedCaptionsError(e, "parse", ccContent);
                 }
             }
         },
 
+        getIsDefault:function(mediaInfo){
+            return mediaInfo.index === this.mediaInfos[0].index; //TODO How to tag default. currently same order as in manifest. Is there a way to mark a text adaptation set as the default one?
+        },
+
         abort:function() {
-            this.getTextTrackExtensions().deleteCues(this.videoModel.getElement());
+            this.videoModel.getElement().textTracks.removeEventListener('change', onTextTrackChange);
+            this.textTrackExtensions.deleteAllTextTracks();
```

Comment:
I had another think about this over the weekend. It seems to me that this is the wrong place to do this stuff. If TextSourceBuffer is a polyfill, abort is supposed to abort the appending of the current segment only (http://www.w3.org/TR/media-source/#widl-SourceBuffer-abort-void). This seems like a reset method.

I know this was here before you got involved, but it's the first time I've noticed it :wink: